lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -8386,23 +8386,35 @@ bool SwiftASTContext::CacheUserImports(SwiftASTContext &swift_ast_context,
   return true;
 }
 
+static std::pair<Module *, lldb::user_id_t>
+GetCUSignature(CompileUnit &compile_unit) {
+  return {compile_unit.GetModule().get(),
+          compile_unit.GetID()};
+}
+
 bool SwiftASTContext::GetCompileUnitImports(
-    SwiftASTContext &swift_ast_context, SymbolContext &sc,
-    lldb::StackFrameWP &stack_frame_wp,
+    SymbolContext &sc, lldb::StackFrameWP &stack_frame_wp,
     llvm::SmallVectorImpl<swift::AttributedImport<swift::ImportedModule>>
-        &modules,
+        *modules,
     Status &error) {
+  CompileUnit *compile_unit = sc.comp_unit;
+  if (compile_unit)
+    if (!m_cu_imports.insert(GetCUSignature(*compile_unit)).second)
+      // List of imports isn't requested and we already processed this CU?
+      if (!modules)
+        return true;
```

Comment:
Not sure if that matters, but if we failed to import modules before this still returns true. From what I can see this could cause that a Swift expression fails the first time but then all following expressions in the same CU pass?